alibi/explainers/integrated_gradients.py

```diff
@@ -761,6 +814,13 @@ def explain(self,
             for each feature.
 
         """
+        # target handling logic
+        if self.target_fn and target is not None:
```

Comment:
another way to look at it is that  `target` takes precedence I am just think that the user might want to change this at explain time without having to create the expaliner object again?

Reply:
That's a good point, I didn't want to complicate the logic, but we could also have `target` override any pre-set `target_fn`, a warning would need to be emitted. Let me think about it.

Reply:
Having detectors/explains which can be modified post-init generally seems like a nice idea (especially if init is expensive), but it does open up a little bit of a question of where to stop with this. For example, for many (or all?) of the offline detectors we could actually have `p_val` as a kwarg of `predict`. Perhaps this is best dealt with by opening new issue(s)?

Reply:
Yea, it's a design discussion. We've had similar discussions in the past, the line between what you would want to change on every `explain` call vs what you want fixed is very fuzzy.